kuzzleio/kuzzle
# Directories overhaul

# Description

Cleanup PR: make the directory structure more consistent with their respective role. Mainly, stop using `lib/api` to store everything.

Exhaustive change list:

* `lib/api/core` => `lib/core` 
* `lib/core/plugins/pluginsXXX.js` => `lib/core/plugins/XXX.js` (e.g. `lib/core/plugins/pluginsManager.js` => `lib/core/plugins/manager.js`)
* `lib/api/controllers/funnelController.js` => `lib/api/funnel.js`
* `lib/api/controllers/routerController.js` => `lib/core/router.js`
* `lib/api/controllers/XXXController.js` => `lib/api/controllers/XXX.js` (e.g. `lib/api/controllers/adminController.js` => `lib/api/controllers/admin.js`)
* `lib/api/kuzzle.js` => `lib/kuzzle.js`
* `lib/kuzzleDebug.js` => `lib/util/debug.js`

Test directories have been updated accordingly.

(jira: KZL-221)

# How to test

Do not read that PR. There is no code change, apart from files being moved around and require directories been updated, and that's that.

Instead, clone this branch, and take a look at the project directories, check if that suits you, if you like it, etc.

# Other changes

A few very old and obsolete README.md files have been deleted, along with 2 obsolete unit-test mock files.
